scripts/convert_hf_checkpoint.py

```diff
@@ -41,58 +58,85 @@ def permute(w):
             .reshape(dim, dim)
         )
 
-    with torch.no_grad():
-        sd["transformer.wte.weight"].copy_(sd_hf["model.embed_tokens.weight"])
-        sd["transformer.ln_f.scale"].copy_(sd_hf["model.norm.weight"])
-        sd["lm_head.weight"].copy_(sd_hf["lm_head.weight"])
-
-        for i in range(n_blocks):
-            sd[f"transformer.h.{i}.attn.c_proj.weight"].copy_(
-                sd_hf[f"model.layers.{i}.self_attn.o_proj.weight"]
-            )
-
-            sd[f"transformer.h.{i}.attn.c_attn.weight"][:qkv_size] = permute(
-                sd_hf[f"model.layers.{i}.self_attn.q_proj.weight"]
-            )
-            sd[f"transformer.h.{i}.attn.c_attn.weight"][qkv_size:-qkv_size] = permute(
-                sd_hf[f"model.layers.{i}.self_attn.k_proj.weight"]
-            )
-            sd[f"transformer.h.{i}.attn.c_attn.weight"][-qkv_size:] = sd_hf[
-                f"model.layers.{i}.self_attn.v_proj.weight"
-            ]
-
-            sd[f"transformer.h.{i}.mlp.c_fc1.weight"].copy_(
-                sd_hf[f"model.layers.{i}.mlp.gate_proj.weight"]
-            )
-            sd[f"transformer.h.{i}.mlp.c_fc2.weight"].copy_(
-                sd_hf[f"model.layers.{i}.mlp.up_proj.weight"]
-            )
-            sd[f"transformer.h.{i}.mlp.c_proj.weight"].copy_(
-                sd_hf[f"model.layers.{i}.mlp.down_proj.weight"]
-            )
-
-            sd[f"transformer.h.{i}.rms_1.scale"].copy_(
-                sd_hf[f"model.layers.{i}.input_layernorm.weight"]
-            )
-            sd[f"transformer.h.{i}.rms_2.scale"].copy_(
-                sd_hf[f"model.layers.{i}.post_attention_layernorm.weight"]
-            )
+    bin_files = set(el for el in bin_index["weight_map"].values())
```

Comment:
nit: move this beside where you load the `bin_index`, above `def permute`

Reply:
Done